resources/todomvc/architecture-examples/svelte/src/App.svelte

```diff
@@ -0,0 +1,121 @@
+<script>
+	import { tick } from 'svelte';
+    import { router } from './router.js';
+    import { uuid } from './utils.js';
+
+    import "./app.css";
+    import "todomvc-app-css/index.css";
+    import "todomvc-common/base/css";
+
+    let currentFilter = "all";
+    let items = [];
+    let editing = null;
+
+    function addItem(event) {
+        const text = event.target.value.trim();
+
+        if (event.key === "Enter") {
+            items = items.concat({
+                id: uuid(),
+                description: text,
+                completed: false,
+            });
+            event.target.value = "";
+        }
+    }
+
+    function removeItem(index) {
+        items = items.slice(0, index).concat(items.slice(index + 1));
+    }
+
+    function updateItem(event) {
+        const text = event.target.value.trim();
+
+        if (text.length === 0) {
+            removeItem(event.target.getAttribute('data-index'));
+        } else {
+            items[editing].description = text;
+        }
+
+        editing = null;
+    }
+
+    function toggleAllItems(event) {
+        items = items.map((item) => ({
+            id: item.id,
+            description: item.description,
+            completed: event.target.checked,
```

Comment:
nit: I'd use 
```js
{
  ...item,
  completed: event.target.checked,
}
```
so that we don't have to change it if for any reason the structure of an item changes in the future.

Reply:
✅